File: Go/factorial.go
package main

import "fmt"


func factorial (i int) int {
    if i == 0 {
        return 1

    } else {
        return i*factorial (i-1);
    }
}


func main() {
    for i := 0; i<10; i++ {
        fmt.Println(factorial(i));
    }
}

File: C/factorial.c
#include <stdio.h>


int factorial(int n)
{
    return n == 0 ? 1 : n * factorial(n-1);
}


int main()
{
    for (int i=0; i<10; ++i)
    {
        printf("Factorial of %d = %d\n", i, factorial(i));
    }
}
